src/mlpack/methods/ann/layer/nearest_interpolation_impl.hpp

```diff
@@ -0,0 +1,228 @@
+/**
+ * @file methods/ann/layer/nearest_interpolation_impl.hpp
+ * @author Abhinav Anand
+ *
+ * Implementation of the NearestInterpolation layer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_NEAREST_INTERPOLATION_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_NEAREST_INTERPOLATION_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "nearest_interpolation.hpp"
+
+namespace mlpack {
+
+template<typename MatType>
+NearestInterpolationType<MatType>::NearestInterpolationType():
+  Layer<MatType>(),
+  inRowSize(0),
+  inColSize(0),
+  outRowSize(0),
+  outColSize(0),
+  depth(0),
+  batchSize(0)
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+NearestInterpolationType<MatType>::
+NearestInterpolationType(const size_t inRowSize,
+                         const size_t inColSize,
+                         const size_t outRowSize,
+                         const size_t outColSize,
+                         const size_t depth) :
+  Layer<MatType>(),
```

Comment:
This is pedantic, but double-indent is preferred for wrapped lines like this initialization list (and others).